client/my-sites/plugins/plugin-activate-toggle/index.jsx

```diff
@@ -1,87 +1,95 @@
-	DisconnectJetpackButton = require( 'my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button' ),
-	analytics = require( 'lib/analytics' );
+import PluginsActions from 'lib/plugins/actions';
+import PluginsLog from 'lib/plugins/log-store';
+import PluginAction from 'my-sites/plugins/plugin-action/plugin-action';
+import DisconnectJetpackButton from 'my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button';
+import { recordGoogleEvent, recordTracksEvent } from 'state/analytics/actions';
 
-module.exports = React.createClass( {
-
-	displayName: 'PluginActivateToggle',
-
-	propTypes: {
-		isMock: React.PropTypes.bool,
-		site: React.PropTypes.object.isRequired,
-		plugin: React.PropTypes.object.isRequired,
-	},
-
-	toggleActivation: function() {
-		if ( this.props.isMock || this.props.disabled ) {
+export class PluginActivateToggle extends Component {
+	toggleActivation = () => {
+		const { isMock, disabled, site, plugin, notices } = this.props;
+		if ( isMock || disabled ) {
 			return;
 		}
 
-		PluginsActions.togglePluginActivation( this.props.site, this.props.plugin );
-		PluginsActions.removePluginsNotices( this.props.notices.completed.concat( this.props.notices.errors ) );
+		PluginsActions.togglePluginActivation( site, plugin );
+		PluginsActions.removePluginsNotices( notices.completed.concat( notices.errors ) );
 
-		if ( this.props.plugin.active ) {
-			analytics.ga.recordEvent( 'Plugins', 'Clicked Toggle Deactivate Plugin', 'Plugin Name', this.props.plugin.slug );
-			analytics.tracks.recordEvent( 'calypso_plugin_deactivate_click', {
-				site: this.props.site.ID,
-				plugin: this.props.plugin.slug
+		if ( plugin.active ) {
+			this.props.recordGoogleEvent( 'Plugins', 'Clicked Toggle Deactivate Plugin', 'Plugin Name', plugin.slug );
+			this.props.recordTracksEvent( 'calypso_plugin_deactivate_click', {
+				site: site.ID,
+				plugin: plugin.slug
 			} );
 		} else {
-			analytics.ga.recordEvent( 'Plugins', 'Clicked Toggle Activate Plugin', 'Plugin Name', this.props.plugin.slug );
-			analytics.tracks.recordEvent( 'calypso_plugin_activate_click', {
-				site: this.props.site.ID,
-				plugin: this.props.plugin.slug
+			this.props.recordGoogleEvent( 'Plugins', 'Clicked Toggle Activate Plugin', 'Plugin Name', plugin.slug );
+			this.props.recordTracksEvent( 'calypso_plugin_activate_click', {
+				site: site.ID,
+				plugin: plugin.slug
 			} );
 		}
-	},
+	}
 
-	render: function() {
-		var inProgress;
+	render() {
+		const { site, plugin, isMock, disabled, translate } = this.props;
 
-		if ( ! this.props.site ) {
+		if ( ! site ) {
 			return null;
 		}
 
-		inProgress = PluginsLog.isInProgressAction( this.props.site.ID, this.props.plugin.slug, [
+		const inProgress = PluginsLog.isInProgressAction( site.ID, plugin.slug, [
 			'ACTIVATE_PLUGIN',
 			'DEACTIVATE_PLUGIN'
 		] );
 
-		if ( this.props.plugin && 'jetpack' === this.props.plugin.slug ) {
+		if ( plugin && 'jetpack' === plugin.slug ) {
```

Comment:
After doing some tests here and there, I confirm that this is not happening. However, I'm still sticking to my opinion here, because it's possible that such cases might occur while we're migrating plugins to Redux. We've seen that happening with the `getSelectedSite()` migration, for example in #9562.